connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -181,7 +184,11 @@ public void execute() {
      */
     private boolean sendRecords() {
         int processed = 0;
-        for (final SourceRecord record : toSend) {
+        for (SourceRecord preTransformRecord : toSend) {
+            final SourceRecord record = transformationChain.apply(preTransformRecord);
+
+            if (record == null) continue;
```

Comment:
That's a good point. It also doesn't seem appropriate to call `commitRecord()` for a record that's not actually being committed. I'm thinking of just documenting this caveat that some records may not get this callbacks at all if users filter with transformations.